src/main/java/graphql/ExecutionInput.java

```diff
@@ -245,27 +268,87 @@ public Builder localContext(Object localContext) {
+         * The legacy context object
          *
          * @param context the context object to use
+         *
          * @return this builder
+         *
+         * @deprecated - use {@link #graphQLContext(GraphQLContext)} instead
          */
+        @Deprecated
         public Builder context(Object context) {
             this.context = context;
             return this;
         }
 
+        /**
+         * The legacy context object
+         *
+         * @param contextBuilder the context builder object to use
+         *
+         * @return this builder
+         *
+         * @deprecated - use {@link #graphQLContext(GraphQLContext.Builder)} instead
+         */
+        @Deprecated
         public Builder context(GraphQLContext.Builder contextBuilder) {
             this.context = contextBuilder.build();
             return this;
         }
 
+        /**
+         * The legacy context object
+         *
+         * @param contextBuilderFunction the context builder function to use
+         *
+         * @return this builder
+         *
+         * @deprecated - use {@link #graphQLContext(UnaryOperator)} instead
+         */
+        @Deprecated
         public Builder context(UnaryOperator<GraphQLContext.Builder> contextBuilderFunction) {
             GraphQLContext.Builder builder = GraphQLContext.newContext();
             builder = contextBuilderFunction.apply(builder);
             return context(builder.build());
         }
 
+        /**
+         * This allows you to set up your own context object that will be passed to all data fetchers during execution
+         *
+         * @param context the context object to use
+         *
+         * @return this builder
+         */
+        public Builder graphQLContext(GraphQLContext context) {
+            this.graphQLContext = assertNotNull(context, () -> "you MUST provide a non null GraphqlContext");
+            return this;
+        }
+
+        /**
+         * This allows you to set up your own context object that will be passed to all data fetchers during execution
+         *
+         * @param contextBuilder the {@link GraphQLContext.Builder} object to use
+         *
+         * @return this builder
+         */
+        public Builder graphQLContext(GraphQLContext.Builder contextBuilder) {
+            return graphQLContext(contextBuilder.build());
+        }
+
+        /**
+         * This allows you to set up your own context object that will be passed to all data fetchers during execution
+         *
+         * @param contextBuilderFunction as callback give a {@link GraphQLContext.Builder} object to use
+         *
+         * @return this builder
+         */
+        public Builder graphQLContext(UnaryOperator<GraphQLContext.Builder> contextBuilderFunction) {
+            GraphQLContext.Builder builder = GraphQLContext.newContext();
+            builder = contextBuilderFunction.apply(builder);
+            return graphQLContext(builder.build());
+        }
```

Comment:
Irrespective of whether `graphQLContext` is an instance that can or cannot be replaced, I think this method should allow operating on the default instance. I can imagine cases where some code only has access to a `GraphQLContext.Builder` and wants to update the context without actually replacing it.

Reply:
This comment does not match any more